subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleExtension.java

```diff
@@ -113,7 +118,29 @@ public File getConfigDir() {
      */
     @Incubating
     public void setConfigDir(File configDir) {
-        this.configDir = configDir;
+        this.configDir.set(configDir);
+    }
+
+    /**
+     * Gets the configuration directory.
+     *
+     * @return The configuration directory
+     * @since 4.7
+     */
+    @Incubating
+    public DirectoryProperty getConfigDirectory() {
+        return configDir;
+    }
+
+    /**
+     * Set the provider for calculating the configuration directory.
+     *
+     * @param configDir Configuration directory provider
+     * @since 4.7
+     */
+    @Incubating
+    public void setConfigurationDirectory(Provider<Directory> configDir) {
```

Comment:
:x: This method shouldn't exist.

Reply:
That makes all sense. However, I am still not clear on how to defer rendering the deprecation message if set through the extension by the user without a `Callable`. Let's just chat tomorrow. That'll get us to the goal faster.

Reply:
You don't need this method for that, you can just call `getConfigurationDirectory.set()` (which is all this method does anyway.

Reply:
I guess I misunderstand one of your previous comments. done.